src/exchange/token_usage_collector.py

```diff
@@ -0,0 +1,40 @@
+from collections import defaultdict
+from dataclasses import dataclass
+import queue
+from typing import List
+
+from exchange.providers.base import Usage
+
+
+@dataclass
+class TokenUsage:
```

Comment:
nit: I'd avoid a new object here and instead make a map of the existing Usage object stored in a dict with model as the key?

Reply:
yes, can do